### ISSUE TITLE
[tune] MLFlow, Neural Network Training (PyTorch-Lightning), and Cross Validation

### ISSUE THREAD

**Issue by andrewredd** [ray-project/ray#11711]
### Describe your feature request

I'm trying to do something that would require repetition similar to that mentioned in: #7744. But I'm having a hard time understanding how I can get the right values piped through the process.

1) I'm running a PyTorch-Lightning Model with the TuneCallback. I've realized that my network is overfitting on my validation when I run hundreds of models. (Good on Validation, Poor test set error). I'd like to run 3 fold CV on each configuration and then average the output. Is there a way to do this in conjunction with HyperBand/BO?

2) I'm using MLFlow and would like to view the performance on the averaged metrics rather than on each individual run. How would I be able to group the sub runs?

I know there is a repeater but I'm unsure how that would interact with the MLFlow Logging setup or the TuneCallback with Pytorch Lightning.

Is there a better way to think about this workflow?

Thanks for any help! I've really been impressed with the framework thus far. This is approaching optimal state for me and my company's data science workflow!
